### DISCUSSION TITLE
Changing the qualification level leads to an error message

### DISCUSSION THREAD

**Discussion by K8Sudo** [monarc-project/MonarcAppFO#332] [Q&A] [answered]
**Describe the bug**
When changing some of the qualification levels, an error message is displayed. The minimum value returned in the popup is always the currently specified value.

**To Reproduce**
1. Go to MyPrintEN
2. Go to Asset: Server management -> threat label: Equipment malfunction or failure -> Vulnerability label: No service level management
3. Change the qualification level from 5 to 1

**Expected behavior**
A change of the qualification level without error message

**Screenshots**
<img width="375" alt="Bildschirmfoto 2020-09-07 um 13 25 24" src="https://user-images.githubusercontent.com/47662401/92383676-12c62380-f10f-11ea-9c96-3d554398dc32.png">
<img width="1440" alt="Bildschirmfoto 2020-09-07 um 13 26 01" src="https://user-images.githubusercontent.com/47662401/92383690-18236e00-f10f-11ea-9ead-97c29fadb46c.png">

**Desktop (please complete the following information):**

| Questions                                | Answers
|-----------------------------------------|--------------------
| Type of issue                           | Bug
| OS version (server)                 | Ubuntu 18.04 LTS
| OS version (client)                   | MacOS, Win10
| PHP version                             | 7.4
| MONARC version / git hash     | 2.9.16
| Browser                                    | Firefox, Chrome, Safari

**Additional context**
The error message appears in both our and the my.monarc instances

**Comment by ruslanbaidan** [comment, marked as answer]
The first thing is to understand that reduction of vulnerability qualification level on treated risks (which have Reduction treatment) means that the situation changed and some controls implemented. Normally it is not suppose to be done this way and performed by recommendations implementation.

The reason of the error message is because you are trying to change vulnerability rate from 5 to 1 (changing to 2 is not possible as well), which means reducing of Vulnerability level by 5 - 1 = 4 points, which is not possible as the level is 2:
![vuln_level_before_change](https://user-images.githubusercontent.com/3246171/92585251-02d84c00-f295-11ea-8016-914463849da6.png)

It is possible to change the vulnerability level to 3, which means the Vulnerability level will be reduced by 5 - 3 = 2 points and the value will be 0:
![vuln_level_after_change](https://user-images.githubusercontent.com/3246171/92585285-0a97f080-f295-11ea-8f50-1747bbecc999.png)
